### PR TITLE
Fixed issue #14310: Google Analytics functionality doesn't work

### PR DESCRIPTION
**Pull request by gabrieljenik**
- Porting logic from 2.06_sp_1.6.0
- Implementing google tracking by surveyId & group number/name
- Added the possibility of setting the "group number/name" from the controller. Used for Print Answers, Public Stats, Registration, Load Survey y Resume Later.

For reference, in LS 2.06:
- Registration and Load Survey pages came out as Welcome
- Print Answers, Public Stats and Resume Later pages came out with the group number, and without the group name (wrongly I think, as there is not group number for those). I mean, it didn't say print answer or anything like that, even though the code seemed to try to do that.

### CODE REVIEW

**Comment by Shnoulle** [comment]
`trackUrlPageName` can be used for something other (for exemple` <title>`) : why not `PageName` or `PageType`

**Comment by gabrieljenik** [comment]
> `trackUrlPageName` can be used for something other (for exemple` <title>`) : why not `PageName` or `PageType`

You mean here ?
> $aData['aSurveyInfo']['trackUrlPageName'] = 'printanswers';

I wan't this variable to obly be used by tracking url and to remain constant.
Not to get mixed with other purposes which later may end up in changing what we use as a tag for tracking.

If some other variable is needed, let's just add it
If someone want to use it on the ` <title>`they can still use it as `trackUrlPageName`

right?

**Review by Shnoulle — APPROVED** [review]
Right :+1:

File: application/core/LS_Twig_Extension.php [hunk 1]
```diff
@@ -587,4 +587,58 @@ public static function getAllTokenAnswers( $iSurveyID )
...
+                } else {
+                    if (
+                        $showgroupinfo == 'both'
+                        || $showgroupinfo == 'name'
+                        || ($showgroupinfo == 'choose' && !isset($survey->showgroupinfo))
+                        || ($showgroupinfo == 'choose' && $survey->showgroupinfo == 'B')
+                        || ($showgroupinfo == 'choose' && $survey->showgroupinfo == 'N')
+                    ) {
+                        $groupInfo = LimeExpressionManager::GetStepIndexInfo($moveInfo['seq']);
+                        $groupName = isset($groupInfo['gname']) ? $groupInfo['gname'] : '';
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/core/LS_Twig_Extension.php:633]: I would say is a new feature.

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/core/LS_Twig_Extension.php:633]: yep (i resolve the conversation)

**Pull request merged**